src/main/scala/org/apache/spark/sql/pulsar/PulsarHelper.scala

```diff
@@ -207,6 +216,33 @@ private[pulsar] case class PulsarHelper(
     }.toMap)
   }
 
+  def latestOffsets(startingOffset: streaming.Offset,
+                    admissionLimits: AdmissionLimits): SpecificPulsarOffset = {
+    // implement helper inside PulsarHelper in order to use getTopicPartitions
+    val latestOffsets = fetchLatestOffsets().topicOffsets
```

Comment:
Can you change to getTopicPartitions? Polling the latest offset here is unnecessary.